libs/go/libp2p_node/dht/dhtclient/dhtclient.go

```diff
@@ -531,7 +532,7 @@ func (dhtClient *DHTClient) RouteEnvelope(envel *aea.Envelope) error {
 	// lookupResponse must be set
 	record := lookupResponse.AgentRecord
```

Comment:
```Accessing field 'lookupResponse.AgentRecord' might lead to a nil pointer dereference```

Reply:
there are more... outside the scope of this PR

Reply:
nice popint, but it's like we trust protobuf here